core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala

```diff
@@ -123,6 +126,23 @@ class SnappySessionState(snappySession: SnappySession)
     }
   }
 
+  /**
+   * This rule sets the flag at query level to link the partitions to
+   * be created for tables to be the same as number of buckets. This will avoid
+   * exchange on one side of a non-collocated join in most cases.
+   */
+  object LinkPartitionsToBuckets extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = {
+      plan.foreach {
+        case j: Join if !planner.asInstanceOf[SnappyStrategies]
```

Comment:
Why only for localjoin? Even for sort merge join the exchange can be avoided if rdd delinking is disabled.